src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -941,33 +941,19 @@ private static bool TrailingZeros(ReadOnlySpan<char> s, int index)
         private static bool NumberBufferToDouble(ref NumberBuffer number, ref double value)
         {
             double d = NumberToDouble(ref number);
-            uint e = DoubleHelper.Exponent(d);
-            ulong m = DoubleHelper.Mantissa(d);
-
-            if (e == 0x7FF)
+            if (!Double.IsFinite(d))
             {
+                value = default;
                 return false;
             }
 
-            if (e == 0 && m == 0)
+            if (d == 0.0)
             {
-                d = 0;
+                d = 0.0;
             }
 
             value = d;
```

Comment:
Maybe:
```C#
value = (d == 0.0) ? 0.0 : d;
```
?

Reply:
@stephentoub indeed 🙂 do we need these extra parentheses?

Reply:
The compiler doesn't... just seemed slightly easier to read than `value = d == 0.0 ? 0.0 : d;`.  No strong preference.